src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -104,7 +110,15 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
                             {
                                 if (listPackageArgs.ReportType != ReportType.Default)  // generic list package is offline -- no server lookups
                                 {
-                                    await GetRegistrationMetadataAsync(packages, listPackageArgs);
+                                    IEnumerable<Lazy<INuGetResourceProvider>> providers = Repository.Provider.GetCoreV3();
+                                    IEnumerable<PackageSource> sources = listPackageArgs.PackageSources;
+                                    foreach (PackageSource source in sources)
+                                    {
+                                        var sourceRepository = Repository.CreateSource(providers, source, FeedType.Undefined);
+                                        _sourceRepositoryCache[(providers, source)] = sourceRepository;
+                                    }
+
+                                    await GetRegistrationMetadataAsync(packages, listPackageArgs, providers);
```

Comment:
Considering that `providers` is the same for every source, I don't understand the benefit of making it part of the dictionary key, and passing it around to the methods that later look it up.  Doesn't `Dictionary<PackageSource, SourceRepository>` work?

Reply:
@zivkan 
Addressed.